modules/tests/linkchecker.py

```diff
@@ -199,25 +199,25 @@ def internal_external_link_checker(filepath, html_str):
                 # If problem detected, add to problem list
                 if links_list[link]:
                     if link not in problems:
-                        problems.append(link)
+                        problems.append(f"[other] {link}")
```

Comment:
@IanDavila, I actually don't know what case this would occur for, but I know it does occur in some cases through testing (the same context as "internal link missing" where I observed it). Do you have any idea what a better descriptor would be here?

Reply:
Since `links_list` is global, it will not try to figure out why it was broken again for a different file, it is likely that it had the same problem. It's used to speed up the script, like a cache. Only bad thing about this current implementation is that it does not store the reasoning, it only stores that it had problems with it (boolean)

Reply:
```
if link not in problems:
```

will no longer work since the problem in `problems` holds the reasoning such as ```problems.append(f"[{r.status_code}] {link}")```

Reply:
TLDR: `problems` is local to the file, and `links_list` is viewed by all files

Reply:
So maybe we want problems to have more structure, maybe a dict `link -> problem` or `link -> [problem, problem]` so that we can check for inclusion via the link and still keep the problem. Probably the array idea is overkill I doubt there would be multiple different detected problems with the same link.